theano/tensor/opt.py

```diff
@@ -1454,13 +1639,8 @@ def local_fill_sink(node):
             new_client.owner.outputs[0].clients = client.outputs[0].clients
             r = local_fill_sink.transform(new_client.owner)
             if not r:
-                all_clients_replaced = False
                 continue
             replacements.update(r)
-        else:
-            all_clients_replaced = False
-    if all_clients_replaced:
-        replacements.pop(node.outputs[0], None)
```

Comment:
To make it more explicit: if adding back those lines do not lead to a speed up, then it is better to remove them.